omics/omics_uri_agent/uri_parser.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+from urllib.parse import urlparse
+from omics.transfer import (
+    ReadSetFileName,
+    ReferenceFileName
+)
+
+
+# Example URIs supported:
+# omics://429915189008.storage.us-east-1.amazonaws.com/1981413158/readSet/5346184667/source1
+# omics://429915189008.storage.us-east-1.amazonaws.com/1981413158/reference/5346184667/source
+class OmicsUri:
+    SCHEME = "omics"
+    NETLOC = "STORAGE"
+    IDENTIFIER = ""
+    DEFAULT_FILE_NAME = None
+
+    def __init__(self, omics_uri):
+        # https://docs.python.org/3/library/urllib.parse.html
+        url_parts = urlparse(omics_uri.upper())
+        if url_parts.scheme != self.SCHEME:
+            raise ValueError(f"Invalid URI scheme, expected {self.SCHEME}")
```

Comment:
Looks like a URI could have multiple validation errors.  You could consider passing back a boolean (passed or failed validation) and a list of validation errors instead of throwing an exception.  That would be more user-friendly.

Reply:
Disagree here. While I agree that having the ability to fix multiple issues at once would be slightly easier for user, I'm not sure that returning a ```false``` boolean instead of an exception is the right behavior. Consider the scenario where a user tries to use the results from the uri parser to run some workflow. If the user does not check the validation boolean before passing in the result of the parser, then they will not know that there's something wrong until they get an omics resource not found error in the middle of their workflow run. IMO fast failing might be a cleaner solution

Reply:
Looks like the `urlparse` does use exceptions to communicate parse errors, but also has a `strict_parsing` flag to turn it off.

Is there a benchmark parsing library that you used as an example for the design?

Reply:
Can you explain what you mean by a benchmark parsing library?

Reply:
An example library that you might have gathered design ideas from, ex: [mackness / Uriparse](https://github.com/mackness/Uriparse)